packages/lib-core/src/types/common.ts

```diff
@@ -0,0 +1,42 @@
+import type { AnyObject } from '@monorepo/common';
+import type { ReactNode } from 'react';
+
+// Type for extension hook
+export type ExtensionHook<TType, TOptions extends AnyObject = AnyObject> = (
+  options: TOptions,
+) => [TType, boolean, unknown];
+
+export type ExtensionK8sResourceIdentifier = {
+  group?: string;
+  version?: string;
+  kind?: string;
+};
+
+export type TypeAndTitle = {
+  /** Type ID for the item type. */
+  type: string;
+  /** Title for the item */
+  title: string | ReactNode;
+};
+
+// Backwards compatible types
+
+export type ExtensionK8sKindVersionModel = ExtensionK8sResourceIdentifier & {
```

Comment:
We can create mapped type helpers, for example:

```ts
type Bar = RequiredProperties<Foo, 'a' | 'b'>; // remove optional modifier for given properties of Foo
type Bar = OptionalProperties<Foo, 'a' | 'b'>; // add optional modifier for given properties of Foo
```

Can be done later as follow-up improvement.